tests/test_main.py

```diff
@@ -1472,3 +1473,71 @@ class Item(BaseModel):
 
     assert id(image_1) == id(item.images[0])
     assert id(image_2) == id(item.images[1])
+
+
+def test_mapping_retains_type_subclass():
+    class Map(dict):
```

Comment:
```suggestion
    class CustomMap(dict):
```
might be a clearer name?